bank_accounts.rb

```diff
@@ -4,13 +4,18 @@ module Bank
   class Account
     attr_reader :balance, :owner, :id, :open_date
     def initialize(id, initial_balance, open_date, owner = nil)
-      if(initial_balance < 0)
-        raise ArgumentError, "Your initial balance must be a positive value."
-      end
-      @balance = initial_balance
       @id = id
+      @balance = initial_balance
+      @min_balance = 0
       @open_date = open_date
       @owner = owner
+      check_balance
+    end
+
+    def check_balance
+      if @balance < @min_balance
+        raise ArgumentError, "Your initial balance must be at least $#{@min_balance/100}."
+      end
     end
 
     def self.all
```

Comment:
I like the way you use the CSV file to pull out the one account match in your `find` method